src/stores/index.js

```diff
@@ -991,10 +976,91 @@ const Store = Reflux.createStore({
     this.state.hasUnsavedChanges = false;
   },
 
+  async _onViewChangedToConnectionForm() {
+    const currentConnection = this.state.currentConnection;
+    const driverUrl = currentConnection.driverUrl;
+    const url = this.state.isURIEditable ? this.state.customUrl : driverUrl;
+    const currentSaved = this.state.connections[currentConnection._id];
+
+    if (!currentSaved && url === driverUrl) {
+      this.state.isHostChanged = true;
+      this.state.isPortChanged = true;
+      this.trigger(this.state);
+
+      return;
+    } else if (url === '') {
+      this.state.isHostChanged = false;
+      this.state.isPortChanged = false;
+      this._clearForm();
+      this._clearConnection();
+
+      return;
+    } else if (!Connection.isURI(url)) {
+      this._clearConnection();
+      this.trigger(this.state);
+
+      return;
+    }
+
+    this.StatusActions.showIndeterminateProgressBar();
+
+    try {
+      const buildConnectionModelFromUrl = promisify(Connection.from);
+      const parsedConnection = await buildConnectionModelFromUrl(url);
+
+      this.StatusActions.done();
+
+      currentConnection.set(this._getPoorAttributes(parsedConnection));
+
+      if (currentSaved) {
+        // If we have SSH tunnel attributes, set them here.
+        if (currentSaved.sshTunnel !== 'NONE') {
+          this._setSshTunnelAttributes(currentSaved, currentConnection);
+        }
+
+        // If we have TLS attributes, set them here.
+        if (currentSaved.sslMethod !== 'NONE') {
+          this._setTlsAttributes(currentSaved, currentConnection);
+        }
+
+        currentConnection.name = currentSaved.name;
+        currentConnection.color = currentSaved.color;
+        currentConnection.lastUsed = currentSaved.lastUsed;
```

Comment:
I think we should look into what we really want to be doing when swapping between the connection form and connection string. I think ideally it shouldn't be doing anything here. Not going to address in this pr.